lib/src/yaml_writer.dart

```diff
@@ -12,25 +12,15 @@ typedef YAMLWriter = YamlWriter;
 class YamlWriter extends Converter<Object?, String> {
   static dynamic _defaultToEncodable(dynamic object) => object.toJson();
 
-  /// The indentation size.
-  ///
-  /// Must be greater or equal to `1`.
-  ///
-  /// Defaults to `2`.
-  final int indentSize;
-
-  /// If `true` it will allow unquoted strings.
-  final bool allowUnquotedStrings;
+  final YamlWriterConfig config;
 
   /// Used to convert objects to an encodable version.
   final Object? Function(dynamic object) toEncodable;
 
-  YamlWriter({
-    int indentSize = 2,
-    this.allowUnquotedStrings = false,
-    Object? Function(dynamic object)? toEncodable,
-  })  : indentSize = max(1, indentSize),
-        toEncodable = toEncodable ?? _defaultToEncodable;
+  const YamlWriter({
+    this.config = const YamlWriterConfig(),
+    this.toEncodable = _defaultToEncodable,
+  });
```

Comment:
How about adding a new factory constructor, `YamlWriter.config`, for these new parameters?  

This way, we can keep the unnamed constructor with the original parameters to avoid breaking the API.

Reply:
Done. And I bump the package version back to 2.2.0.